maven/impl-maven-embedded/src/main/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/DistributionStageImpl.java

```diff
@@ -80,19 +83,31 @@ public boolean accept(File file) {
         return extracted[0];
     }
 
+    private File createTempFile(File toExtract) {
+        try {
+            final File tempFile = Files.createTempFile(toExtract.toPath(), "temp", UUID.randomUUID().toString()).toFile();
```

Comment:
The name should have some specific name - eg. `extractionIsBeingProcessed` to make it clear what it is good for